### ISSUE TITLE
"Hang Closing" option doesn't work.

### ISSUE THREAD

**Issue by Asday** [PyCQA/pycodestyle#846]
I originally posted this to lgeiger/ide-python (issue [#125](https://github.com/lgeiger/ide-python/issues/125#event-2152576921)), but it was closed, and I was told to repost here.  Below is a reproduction of the original issue text.

I can't seem to get closing brackets to outdent automatically using this package.

```python
a = {
    1: 2,
    3: 4,
    }
```

When I type that, and press Enter after the fourth line, the leading whitespace should be removed when I have the "Hang Closing" option checked, but it doesn't.

Atom version: 1:27.2
ide-python version: 1.0.0

**Comment by asottile** [comment]
It's not clear to me how this is related to pycodestyle which just lints code but does not do formatting

@lgeiger maybe you can elaborate on why this is a pycodestyle issue

**Comment by Asday** [comment]
I'm sorry, I don't know.  I'm just some doofus who wants brackets to move about when he presses Enter.  :(

**Comment by JonathanRRogers** [comment]
The IDE-python package can use either [autopep8](https://github.com/hhatto/autopep8) or [YAPF](https://github.com/google/yapf) to format code, so I'd guess this should be an issue with one of those.

**Comment by Asday** [comment]
@lgeiger if you could let me know which of those packages I should raise the issue with, I'd appreciate it.

**Comment by lgeiger** [comment]
@Asday Sorry, I was in a rush and misread your issue. I though it was about the linting. This obviously has nothing todo with pycodestyle, sorry about the noise.
